homeassistant/components/synology_srm/device_tracker.py

```diff
@@ -71,14 +99,31 @@ def scan_devices(self):
         """Scan for new devices and return a list with found device IDs."""
         self._update_info()
 
-        return [device["mac"] for device in self.last_results]
+        return [device["mac"] for device in self.devices]
+
+    def get_extra_attributes(self, device) -> dict:
+        """Get the extra attributes of a device."""
+        device = next(
+            iter([result for result in self.devices if result["mac"] == device]), None
+        )
+        if device:
+            filtered_attributes = {}
+            for attribute, value in device.items():
+                if attribute in ATTRIBUTE_ALIAS:
+                    filtered_attributes[
+                        (
+                            ATTRIBUTE_ALIAS[attribute]
+                            if ATTRIBUTE_ALIAS[attribute]
+                            else attribute
+                        )
+                    ] = value
+            return filtered_attributes
+        return {}
```

Comment:
```suggestion
        filtered_attributes = {}
        if not device:
            return filtered_attributes
        for attribute, alias in ATTRIBUTE_ALIAS.items():
            value = device.get(attribute)
            if value is None:
                continue
            attr = alias or attribute
            filtered_attributes[attr] = value
        return filtered_attributes
```

Reply:
Will do this with the other changes when I get home tonight :)